test/ApiCheck.Test/ReportGenerationTests.cs

```diff
@@ -0,0 +1,1080 @@
+
+            // Assert
+            Assert.NotNull(report);
+            Assert.NotNull(report.Types);
+            var type = Assert.Single(report.Types, t => t.Id == "public class Scenarios.MethodTypesClass");
+            var method = Assert.Single(type.Members, m => m.Id == "public System.Void ParameterlessVoidReturningMethod()");
+        }
+
+        [Fact]
+        public void DetectsProtectedIntReturningMethod()
+        {
+            // Arrange
+            var generator = CreateGenerator(V1Assembly);
+
+            // Act
+            var report = generator.GenerateBaseline();
+
+            // Assert
+            Assert.NotNull(report);
+            Assert.NotNull(report.Types);
+            var type = Assert.Single(report.Types, t => t.Id == "public class Scenarios.MethodTypesClass");
+            var method = Assert.Single(type.Members, m => m.Id == "protected System.Int32 ProtectedIntReturningMethod()");
+        }
+
+        [Fact]
+        public void ProtectedInternalStringReturningMethodWithStringParameter()
+        {
+            // Arrange
+            var generator = CreateGenerator(V1Assembly);
+
+            // Act
+            var report = generator.GenerateBaseline();
+
+            // Assert
+            Assert.NotNull(report);
+            Assert.NotNull(report.Types);
+            var type = Assert.Single(report.Types, t => t.Id == "public class Scenarios.MethodTypesClass");
+            var method = Assert.Single(type.Members, m => m.Id == "protected internal System.String ProtectedInternalStringReturningMethodWithStringParameter(System.String stringParameter)");
+        }
+
+        [Fact]
+        public void InternalClassReturningMethodWithOptionalStringParameter()
+        {
+            // Arrange
+            var generator = CreateGenerator(V1Assembly);
+
+            // Act
+            var report = generator.GenerateBaseline();
+
+            // Assert
+            Assert.NotNull(report);
+            Assert.NotNull(report.Types);
+            var type = Assert.Single(report.Types, t => t.Id == "public class Scenarios.MethodTypesClass");
+            var method = Assert.Single(type.Members, m => m.Id == "internal Scenarios.MethodTypesClass InternalClassReturningMethodWithOptionalStringParameter(System.String defaultParameter = \"hello\")");
+        }
+
+        [Fact]
+        public void PrivateBoolReturningMethodWithOptionalCharParameter()
+        {
+            // Arrange
+            var generator = CreateGenerator(V1Assembly);
+
+            // Act
+            var report = generator.GenerateBaseline();
+
+            // Assert
+            Assert.NotNull(report);
+            Assert.NotNull(report.Types);
+            var type = Assert.Single(report.Types, t => t.Id == "public class Scenarios.MethodTypesClass");
+            var method = Assert.Single(type.Members, m => m.Id == "private System.Boolean PrivateBoolReturningMethodWithOptionalCharParameter(System.Char charParameter = 'c')");
```

Comment:
Classification is separated from filtering on the tool. We want to make sure all elements are categorized correctly independently of whether we filter them or not.